src/citrine/_serialization/properties.py

```diff
@@ -655,11 +738,31 @@ def _deserialize(self, data: dict) -> typing.Any:
             raise AttributeError("Tried to deserialize to {!r}, which has no fields and is not an"
                                  " explicitly serializable class".format(self.klass))
 
-        instance = self.klass.__new__(self.klass)
+        values = {}
+        init_props = set()
         for property_name, field in self.fields.items():
+            if field.use_init:
+                init_props.add(property_name)
             if field.deserializable:
-                value = field.deserialize_from_dict(data)
-                setattr(instance, property_name, value)
+                values[property_name] = field.deserialize_from_dict(data)
+            elif field.default is not None:
+                values[property_name] = field.default
+
+        if len(init_props) > 0:
+            try:
+                instance = self.klass(**{k: v for k, v in values.items() if k in init_props})
+            except TypeError as e:
+                if "required" in str(e) and "argument" in str(e):
```

Comment:
Is there a way to make this based on the specific error class, as opposed to string-matching on the message?

Reply:
Error condition is that you've failed to pass a required argument to __init__.  This fires off `TypeError` with messges of the form 
`TypeError: CLASS.__init__() missing 1 required positional argument: 'arg_name'`
or 
`TypeError: CLASS.__init__() missing 1 required keyword-only argument: 'arg_name'`
I thought just catching a TypeError would catch too many other potential issues.  There's no tighter error type to filter by.

Reply:
What I'm wondering is if we could change those from being TypeErrors to being some other type and then recognize that?

Reply:
We'd need to scan the __init__ argument list and compare what we're about to pass it.  This is actually not terrible because python provides [https://docs.python.org/3.10/library/inspect.html#inspect.getfullargspec] and [https://docs.python.org/3.10/library/inspect.html#inspect.signature], but I, again, trying to maintain a bit of velocity.

Reply:
Test implemented with [signature](https://docs.python.org/3.10/library/inspect.html#inspect.signature).